src/wagtail_analytics/views.py

```diff
@@ -80,28 +83,43 @@ def get_context_data(self, **kwargs):
 
         analytics_settings = AnalyticsSettings.for_site(site)
 
-        if not analytics_settings.google_analytics_view_id:
-            raise Http404("No view id found")
-
         # Show a site switcher form if there are multiple sites
         site_switcher = None
         if Site.objects.count() > 1:
             site_switcher = SiteSwitchForm(site, Site)
 
         context.update(
             {
-                "access_token": get_access_token_from_string(settings.GA_KEY_CONTENT),
-                "id": analytics_settings.google_analytics_view_id,
                 "site": site,
                 "site_switcher": site_switcher,
-                "config_url": reverse("wagtail-analytics-config"),
+                "report_url": reverse(
+                    "wagtail-analytics-report", kwargs={"site_id": site.id}
+                ),
             }
         )
         return context
 
 
-class ConfigView(View):
+class AnalyticsReportView(View):
     def get(self, request, *args, **kwargs):
-        page_id = kwargs.get("page_id", None)
-        access_token = get_access_token_from_string(settings.GA_KEY_CONTENT)
-        return JsonResponse({"access_token": access_token, "page_id": page_id})
+        site_id = kwargs.get("site_id", None)
+        site = get_object_or_404(Site, id=site_id)
+        analytics_settings = AnalyticsSettings.for_site(site)
+        if analytics_settings.plausible_enabled is True:
+            client = PlausibleAPIClient(
+                site.hostname,
+                wagtail_analytics_settings.PLAUSIBLE_API_KEY,
+            )
+            report = client.get_report()
+            report_dict = dataclasses.asdict(report)
+            return JsonResponse(report_dict)
+        if analytics_settings.google_analytics_enabled is True:
+            credentials = json.loads(wagtail_analytics_settings.GA_KEY_CONTENT)
+            client = GoogleAnalyticsAPIClient(
+                analytics_settings.google_analytics_property_id,
+                credentials,
+            )
+            report = client.get_report()
+            report_dict = dataclasses.asdict(report)
+            return JsonResponse(report_dict)
+        return JsonResponse({"error": "No analytics provider enabled"})
```

Comment:
Better is to raise a 404 here (or return explicitly the 404 status code also), so we can act on it in our javascript